context/src/repo/validator.rs

```diff
@@ -85,6 +86,10 @@ pub enum RepoValidationIssueSubOptimal {
 pub enum RepoValidationIssueInvalid {
     #[error("repo branch name too short")]
     RepoBranchNameTooShort(String),
+    #[error("repo sha too short")]
+    RepoShaTooShort(String),
+    #[error("repo sha too long, truncated to {}", MAX_SHA_FIELD_LEN)]
+    RepoShaTooLong(String),
```

Comment:
Note that by saying these errors are 'invalid,' we're saying that, if they're present, we cannot process the bundle for flaky tests and will skip the bundle in processing. Is that the case? Or can we still process the bundle? If we can still process the bundle I'd suggest making these errors 'suboptimal-level' (`RepoValidationIssueSubOptimal`) meaning we'll still track them in the defects table and flag them to the end user, but we can still process the data

Reply:
Right now, we are planning on dropping uploads without a git state, so that makes not having a git state an `INVALID` error - https://trunk-io.slack.com/archives/C044VBASZ0D/p1733183476781069?thread_ts=1733182342.660259&cid=C044VBASZ0D

Reply:
@katchao - Chatted with @pv72895; sounds like we want to keep `RepoShaTooShort` as an `INVALID` error due to our db model's dependency on its presence, but switch `RepoShaTooLong` to a `SUBOPTIMAL` error